File: app/src/main/java/com/kongpf8848/dmail/oauth/hotmail/HotmailApi.java
package com.kongpf8848.dmail.oauth.hotmail;

import com.kongpf8848.dmail.oauth.AuthorizationHeader;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Header;

public interface HotmailApi {

    @GET("/v5.0/me")
    Single<HotmailProfile> profile(@Header("Authorization") AuthorizationHeader header);
}

File: app/src/main/java/com/kongpf8848/dmail/login/oauth/hotmail/HotmailApi.kt
package com.kongpf8848.dmail.login.oauth.hotmail

import com.kongpf8848.dmail.login.oauth.AuthorizationHeader
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Header

interface HotmailApi {
    @GET("/v5.0/me")
    fun profile(@Header("Authorization") header: AuthorizationHeader?): Single<HotmailProfile>
}